module/zfs/zfs_ioctl.c

```diff
@@ -2521,6 +2521,45 @@ zfs_prop_set_special(const char *dsname, zprop_source_t source,
 	return (err);
 }
 
+static boolean_t
+zfs_is_namespace_prop(zfs_prop_t prop)
+{
+	switch (prop) {
+
+	case ZFS_PROP_ATIME:
+	case ZFS_PROP_RELATIME:
+	case ZFS_PROP_DEVICES:
+	case ZFS_PROP_EXEC:
+	case ZFS_PROP_SETUID:
+	case ZFS_PROP_READONLY:
+	case ZFS_PROP_XATTR:
+	case ZFS_PROP_NBMAND:
+		return (B_TRUE);
+
+	default:
+		return (B_FALSE);
+	}
+}
+
+/* Update the VFS's cache of mountpoint properties */
+static void
+update_mount_cache(const char *dsname)
+{
+#ifdef __FreeBSD__
```

Comment:
We try to avoid platform ifdefs as much as possible. This should go in module/os/*/zfs/ instead.